src/common/CircleGameObject.hpp

```diff
@@ -3,18 +3,26 @@
 
 #include <vector>
 #include <memory>
+#include <iostream>
 
 #include "GameObject.hpp"
 #include "../common/Circle.hpp"
+#include "../common/CircleRigidBody.hpp"
+#include "../common/PhysicsController.hpp"
 
 class CircleGameObject : public GameObject {
-    public:
-        CircleGameObject(unsigned int id, sf::Vector2f position, float size, sf::Color color);
-        std::shared_ptr<Circle> getCircle();
-        static std::vector<std::shared_ptr<Circle>> getCircles(
-            std::vector<std::shared_ptr<CircleGameObject>>& circle_game_objects);
-    protected:
-        std::shared_ptr<Circle> mCircle;
+ public:
+     CircleGameObject(unsigned int id, sf::Vector2f position, float radius, sf::Color color,
+        std::shared_ptr<PhysicsController> pc);
+     std::shared_ptr<Circle> getCircle();
+     void setActive(bool is_active);
+     void setVelocity(sf::Vector2f velocity);
+     void setRadius(float radius);
+     void setPosition(sf::Vector2f position);
+     void matchRigid();
+ protected:
+     std::shared_ptr<Circle> mCircle;
```

Comment:
what is this actually shared with?

Reply:
I mean (mCircle)

Reply:
I made this "shared" because there is this function in `CircleGameObject`
```
std::shared_ptr<Circle> CircleGameObject::getCircle() {
    return mCircle;
}
```